go/libkb/socket.go

```diff
@@ -92,6 +94,7 @@ func (g *GlobalContext) GetSocket(clearError bool) (net.Conn, rpc.Transporter, b
 	if sw.err != nil && clearError {
 		g.SocketWrapper = nil
 	}
+	err = sw.err
```

Comment:
you can skip this and just return sw.err...it will still work in the defer.

https://play.golang.org/p/nCeiu3QFLb